File: spec/php/SwitchManualEnumTest.php
<?php
namespace Kaitai\Struct\Tests;

class SwitchManualEnumTest extends TestCase {
    public function testSwitchManualEnum() {
        $r = SwitchManualEnum::fromFile(self::SRC_DIR_PATH . "/switch_opcodes.bin");

        $this->assertEquals(4, count($r->opcodes));

        $this->assertEquals(SwitchManualEnum\Opcode\CodeEnum::STRVAL, $r->opcodes[0]->code);
        $this->assertEquals('foobar', $r->opcodes[0]->body->value);

        $this->assertEquals(SwitchManualEnum\Opcode\CodeEnum::INTVAL, $r->opcodes[1]->code);
        $this->assertEquals(0x42, $r->opcodes[1]->body->value);

        $this->assertEquals(SwitchManualEnum\Opcode\CodeEnum::INTVAL, $r->opcodes[2]->code);
        $this->assertEquals(0x37, $r->opcodes[2]->body->value);

        $this->assertEquals(SwitchManualEnum\Opcode\CodeEnum::STRVAL, $r->opcodes[3]->code);
        $this->assertEquals('', $r->opcodes[3]->body->value);
    }
}

File: spec/zig/tests/switch_manual_enum_test.zig
const _imp_std = @import("std");
const _imp_kaitai_struct = @import("kaitai_struct");
const _imp_switch_manual_enum = @import("../formats/switch_manual_enum.zig");

test "SwitchManualEnum" {
    const file = try _imp_std.fs.cwd().openFile("../../src/switch_opcodes.bin", .{});
    defer file.close();
    var buffer: [8]u8 = undefined;
    var reader = file.reader(&buffer);
    const allocator = _imp_std.testing.allocator;
    var arena = _imp_std.heap.ArenaAllocator.init(allocator);
    defer arena.deinit();
    var _io = _imp_kaitai_struct.KaitaiStream.fromFileReader(&reader);
    const r = try _imp_switch_manual_enum.SwitchManualEnum.create(&arena, &_io, null, null);
    try _imp_std.testing.expectEqual(4, r.opcodes.items.len);
    try _imp_std.testing.expectEqual(_imp_switch_manual_enum.SwitchManualEnum.Opcode.CodeEnum.strval, r.opcodes.items[0].code);
    try _imp_std.testing.expectEqualStrings("foobar", r.opcodes.items[0].body.?.strval.value);
    try _imp_std.testing.expectEqual(_imp_switch_manual_enum.SwitchManualEnum.Opcode.CodeEnum.intval, r.opcodes.items[1].code);
    try _imp_std.testing.expectEqual(66, r.opcodes.items[1].body.?.intval.value);
    try _imp_std.testing.expectEqual(_imp_switch_manual_enum.SwitchManualEnum.Opcode.CodeEnum.intval, r.opcodes.items[2].code);
    try _imp_std.testing.expectEqual(55, r.opcodes.items[2].body.?.intval.value);
    try _imp_std.testing.expectEqual(_imp_switch_manual_enum.SwitchManualEnum.Opcode.CodeEnum.strval, r.opcodes.items[3].code);
    try _imp_std.testing.expectEqualStrings("", r.opcodes.items[3].body.?.strval.value);
}
